### ISSUE TITLE
EXC_BAD_ACCESS startChat

### ISSUE THREAD

**Issue by cancan101** [zendesk/zendesk_sdk_chat_ios#31]
I call `[ZDCChat startChat:nil];`

![image](https://cloud.githubusercontent.com/assets/51059/11045293/b45d5676-86f4-11e5-9629-17508511b3a1.png)

```
1   0x119c6edce StartWebThread()
2   0x112160eaf __pthread_once_handler
3   0x112175f13 _os_once
4   0x112160e4e pthread_once
5   0x118c6d23e +[WebView(WebPrivate) enableWebThread]
6   0x118c675f6 WebKitInitialize
7   0x11025df61 ___UIApplicationLoadWebKit_block_invoke
8   0x111e3049b _dispatch_client_callout
9   0x111e1be28 dispatch_once_f
10  0x10db4fbff _class_initialize
11  0x10db55d23 lookUpImpOrForward
12  0x10db648bb objc_msgSend
13  0x10d2dfe3f -[ZDCJSWidgetIO setupUserAgent]
14  0x10d2dfda7 -[ZDCJSWidgetIO initWithDelegate:andReachability:]
15  0x10d2f01bd -[ZDCStandardChatSession init]
16  0x10d331a5f -[ZDCChat init]
17  0x10d3316e5 __19+[ZDCChat instance]_block_invoke
18  0x111e3049b _dispatch_client_callout
19  0x111e1be28 dispatch_once_f
20  0x10d3316bd +[ZDCChat instance]
21  0x10d33182f +[ZDCChat startChat:]
22  0x10d212366 -[ChatBridge addEvent]
23  0x10e83c85c __invoking___
24  0x10e83c6ae -[NSInvocation invoke]
25  0x10e8cd016 -[NSInvocation invokeWithTarget:]
26  0x10d235fd1 -[RCTModuleMethod invokeWithBridge:module:arguments:]
27  0x10d26e924 -[RCTBatchedBridge _handleRequestNumber:moduleID:methodID:params:]
28  0x10d26de9a __33-[RCTBatchedBridge handleBuffer:]_block_invoke
29  0x111e0fef9 _dispatch_call_block_and_release
30  0x111e3049b _dispatch_client_callout
31  0x111e16988 _dispatch_queue_drain
```

**Comment by cancan101** [comment]
It looks like this is caused by making the call from a thread other than the main IOS thread.

**Comment by oarrabi** [comment]
Hello @cancan101,

`[ZDCChat startChat:nil];` starts the main chat view controller. Since this `[ZDCChat startChat:nil];` manipulates the UI it has to be executed on the main thread.

Is there a reason for trying to call it on another thread other than the UI thread?

**Comment by cancan101** [comment]
React native will by default call from another thread. There is a a work around but perhaps there should be guard agaisnt this / error message

**Comment by schlan** [comment]
Hi @cancan101, thank you for your suggestion, I added it to our backlog.

**Comment by ndobri** [comment]
@cancan101 this has been addressed in Chat SDK [1.3](https://github.com/zendesk/zendesk_sdk_chat_ios/releases/tag/1.3.0.1). Closing this issue now, but of course, you can reopen it if the problem persists.
